form-validation.js

```diff
@@ -7,25 +7,29 @@ const messageMobile = document.querySelector('#messageEmail-mobile');
 const nameInput = document.querySelector('.name');
 const emailInput = document.querySelector('.email');
 const messageInput = document.querySelector('.message');
-// const formbutton = document.querySelector('#form-button');
-// const formbuttondesktop = document.querySelector('#form-button-desk');
+const rawData = localStorage.getItem('cachedFormData');
 
 const formobject = {
   name: '',
   email: '',
   message: '',
 };
 
-function storeInput() {
-  const stringobject = JSON.stringify(formobject);
-  nameInput.value = formobject.name;
+if (rawData) {
+  const formobject = JSON.parse(rawData);
   emailInput.value = formobject.email;
+  nameInput.value = formobject.name;
   messageInput.value = formobject.message;
-  localStorage.setItem('formobject', stringobject);
 }
 
-form.addEventListener('click', storeInput());
-mobileForm.addEventListener('click', storeInput());
+const dataStore = (event) => {
+  const element = event.target;
+  formobject[element.name] = element.value;
+  localStorage.setItem('cachedFormData', JSON.stringify(formobject));
+};
```

Comment:
- [ ] Please from the project requirement, you should show your error message near your submit button so I think you should do that.
![image](https://user-images.githubusercontent.com/92303891/218285825-c653cb75-ca38-4e5f-987d-3fc016fe099e.png)

- [ ] Also, I think you should make your error message specific.

Reply:
where exactly? under? on top? @Owusu-Desmond

Reply:
Hi @stino-x , it can be under the button or on top. please just make it near the button as requested in the project requirement.